gradoop-common/pom.xml

```diff
@@ -20,11 +20,37 @@
         <plugins>
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-shade-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <phase>package</phase>
+                        <goals>
+                            <goal>shade</goal>
+                        </goals>
+                        <configuration>
+                            <minimizeJar>true</minimizeJar>
+                            <artifactSet>
+                                <includes>
+                                    <!-- include BSON object ids -->
+                                    <include>org.mongodb:mongo-java-driver</include>
```

Comment:
did you check the size of the resulting jar? if it includes the whole depedency, maybe the [minimizeJar](https://maven.apache.org/plugins/maven-shade-plugin/shade-mojo.html#minimizeJar) setting is an option?